### PR TITLE
rm public Tick Spacing limits

### PR DESCRIPTION
**Pull request by wjmelements**
Reviewer @hensha256 @snreynolds
There are two reasons to remove unnecessary methods.
1. to save gas on all methods by reducing the size of the 4byte abi lookup table.
2. to reduce codesize
#### Changes
Remove public constant methods MIN_TICK_SPACING and MAX_TICK_SPACING

### CODE REVIEW

File: .forge-snapshots/poolManager bytecode size.snap [hunk 1]
```diff
@@ -1 +1 @@
-24476
\ No newline at end of file
+24339
```
Comment (wjmelements) [inline .forge-snapshots/poolManager bytecode size.snap:1]: decent progress, but still above 24k

**Comment by wjmelements** [comment]
These constants have been public since https://github.com/Uniswap/v4-core/pull/71. I don't think there is a good reason for them to be public.

**Comment by wjmelements** [comment]
The test failure is the same codecoverage 403 as before. I've been told it is fine and expected.

File: .forge-snapshots/poolManager bytecode size.snap (hunk 1 above)
Comment (wjmelements) [reply .forge-snapshots/poolManager bytecode size.snap:1]: This saves 137 bytes. For comparison, using the internal function `checkNotDelegateCall` for `noDelegateCall` saves 149 bytes.

**Review by snreynolds — COMMENTED** [review]
Nice! I like it, tagging @hensha256 also for a review

File: .forge-snapshots/addLiquidity CA fee.snap [hunk 2]
```diff
@@ -1 +1 @@
-170956
\ No newline at end of file
+170604
```
Comment (hensha256) [inline .forge-snapshots/addLiquidity CA fee.snap:1]: wowww nice

**Review by hensha256 — DISMISSED** [review]

**Comment by hensha256** [comment]
Hey @wjmelements ! Thanks for this! Please can you sign your commits so we can merge it?

**Comment by wjmelements** [comment]
> Please can you sign your commits so we can merge it?

That wasn't necessary previously. Can it be skipped? Otherwise I'd have to force-push amended commits.

**Comment by wjmelements** [comment]
ok I have force pushed a signed commit

**Comment by wjmelements** [comment]
@hensha256 please merge

**Review by hensha256 — APPROVED** [review]

**Pull request merged**